test/hosting/test_HostingBasic.cs

```diff
@@ -139,5 +139,37 @@ public static void TestCommandFromNative()
                 File.Delete(target);
             }
         }
+
+        [Fact]
+        public static void TestAddType()
+        {
+            string code = @"
+                using System;
+                public class Foo
+                {
+                    public Foo(string name, string path)
+                    {
+                        this.Name = name;
+                        this.Path = path;
+                    }
+
+                    public string Name;
+                    public string Path;
+                }
+            ";
+
+            using (System.Management.Automation.PowerShell ps = System.Management.Automation.PowerShell.Create())
+            {
+                ps.AddCommand("Add-Type").AddParameter("TypeDefinition", code).Invoke();
+                ps.Commands.Clear();
+
+                var results = ps.AddScript("[Foo]::new('Joe', 'Unknown')").Invoke();
+                Assert.Single(results);
+
+                dynamic foo = results[0];
```

Comment:
The same question.

Reply:
Same here. Also, I have to use `dynamic` to be able to access the properties of the underlying .NET type directly.